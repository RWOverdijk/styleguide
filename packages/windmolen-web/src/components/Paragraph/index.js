import Base from '../Base';

const Paragraph = Base.withComponent('p');

export default Paragraph;
