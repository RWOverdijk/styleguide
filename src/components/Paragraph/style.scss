/** @define Paragraph */

@import "../../css/variables";
@import "../../css/mixins";

.Van-Paragraph {
  @include color-fill($color-dark);

  font-weight: $font-weight-light;
  font-size: 18px;
  line-height: 26px;

  @media (max-width: $container-sm) {
    font-size: 16px;
    line-height: 22px;
  }
}
