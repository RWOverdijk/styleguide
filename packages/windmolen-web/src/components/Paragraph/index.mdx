---
name: Paragraph
menu: Base
route: /Base/Paragraph
---

import { Playground, PropsTable } from 'docz'
import Wrapper from '../Wrapper';
import Paragraph from './'

# Paragraph

## Basic usage
<Playground wrapper={Wrapper}>
  <Paragraph>
    Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
  </Paragraph>
</Playground>
