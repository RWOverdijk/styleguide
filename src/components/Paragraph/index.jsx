import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './style.css';

class Paragraph extends PureComponent {
  getClass(className) {
    let classes = ['Van-Paragraph'];

    if (this.props.className) {
      classes.push(this.props.className);
    }

    classes.push(className);

    return classes.join(' ');
  }

  render() {
    return <p className={this.getClass()}>{this.props.children}</p>;
  }
}

Paragraph.propTypes = {
  className: PropTypes.string
};

export default Paragraph;
