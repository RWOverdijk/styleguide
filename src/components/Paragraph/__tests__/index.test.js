import React from 'react';
import renderer from 'react-test-renderer';
import Paragraph from '../';

describe("<Paragraph>", () => {
  it("renders Paragraph", () => {
    const tree = renderer.create(
      <Paragraph>My Paragraph</Paragraph>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Paragraph className="my-class">My Paragraph</Paragraph>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <Paragraph data-id="0">My Paragraph</Paragraph>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
