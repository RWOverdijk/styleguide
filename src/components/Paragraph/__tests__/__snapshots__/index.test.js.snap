// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph> renders Paragraph 1`] = `
<p
  className="Van-Paragraph"
>
  My Paragraph
</p>
`;

exports[`<Paragraph> renders with specific attributes 1`] = `
<p
  className="Van-Paragraph"
  data-id="0"
>
  My Paragraph
</p>
`;

exports[`<Paragraph> renders with specific className 1`] = `
<p
  className="Van-Paragraph my-class"
>
  My Paragraph
</p>
`;
